linear_operator/operators/block_diag_linear_operator.py

```diff
@@ -68,23 +69,25 @@ def _add_batch_dim(self, other):
         return other
 
     @cached(name="cholesky")
-    def _cholesky(self, upper=False):
+    def _cholesky(
+        self: Float[LinearOperator, "*batch N N"], upper: Optional[bool] = False
+    ) -> Float[LinearOperator, "*batch N N"]:
         from .triangular_linear_operator import TriangularLinearOperator
 
         chol = self.__class__(self.base_linear_op.cholesky(upper=upper))
         return TriangularLinearOperator(chol, upper=upper)
 
-    def _cholesky_solve(self, rhs, upper: bool = False):
+    def _cholesky_solve(self, rhs, upper: Optional[bool] = False) -> Union[LinearOperator, Tensor]:
```

Comment:
Type annotation for `rhs`

Reply:
Looks good.

Reply:
OK. I have changed this to a more detailed definition:
```
def _cholesky_solve(self: Float[LinearOperator, "*batch N N"],
                        rhs: Float[LinearOperator, "batch N M"],
                        upper: Optional[bool] = False) \
            -> Union[Float[LinearOperator, "batch N M"], Float[Tensor, "batch N M"]]:
```